### PR TITLE
NIFI-4080: Added EL support to fields in ValidateCSV

### PR DESCRIPTION
**Pull request by mattyb149**
Thank you for submitting a contribution to Apache NiFi.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [x] Does your PR title start with NIFI-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [x] Has your PR been rebased against the latest commit within the target branch (typically master)?

- [x] Is your initial contribution a single, squashed commit?

### For code changes:
- [x] Have you ensured that the full suite of tests is executed via mvn -Pcontrib-check clean install at the root nifi folder?
- [x] Have you written or updated unit tests to verify your changes?
- [x] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [x] If applicable, have you updated the LICENSE file, including the main LICENSE file under nifi-assembly?
- [x] If applicable, have you updated the NOTICE file, including the main NOTICE file found under nifi-assembly?
- [x] If adding new Properties, have you added .displayName in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [x] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/ValidateCsv.java [hunk 1]
```diff
@@ -227,9 +231,9 @@ public void setPreference(final ProcessContext context) {
         // input is transferred over to Java as is. So when you type the characters "\"
         // and "n" into the UI the Java string will end up being those two characters
         // not the interpreted value "\n".
-        final String msgDemarcator = context.getProperty(END_OF_LINE_CHARACTER).getValue().replace("\\n", "\n").replace("\\r", "\r").replace("\\t", "\t");
-        this.preference.set(new CsvPreference.Builder(context.getProperty(QUOTE_CHARACTER).getValue().charAt(0),
-                context.getProperty(DELIMITER_CHARACTER).getValue().charAt(0), msgDemarcator).build());
+        final String msgDemarcator = context.getProperty(END_OF_LINE_CHARACTER).evaluateAttributeExpressions().getValue().replace("\\n", "\n").replace("\\r", "\r").replace("\\t", "\t");
+        this.preference.set(new CsvPreference.Builder(context.getProperty(QUOTE_CHARACTER).evaluateAttributeExpressions().getValue().charAt(0),
+                context.getProperty(DELIMITER_CHARACTER).evaluateAttributeExpressions().getValue().charAt(0), msgDemarcator).build());
```
Comment (patricker) [inline nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/ValidateCsv.java]: What do you think of waiting until `onTrigger` to build the CsvPreference so you can also evaluate the FlowFile's attributes when you call `evaluateAttributeExpressions`? Then each you can use broader scoped variables, or variables specified at the FlowFile leve.

Comment (mattyb149) [reply nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/ValidateCsv.java]: That would be more useful, this was initially just a quick addition. There is something funky with unit tests when you call isExpressionLanguagePresent() from a custom validator, it throws an NPE. I will look into that and try to move it into onTrigger(). However if for some reason there is a property that must be evaluated early, or too much overhead in building the CsvPreference (not saying there is, just saying if there is), then I'd rather have all the properties either support flow file attributes or none, how does that sound?

Comment (patricker) [reply nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/ValidateCsv.java]: No concerns.

**Comment by mgaido91** [comment]
@mattyb149 there is a test failure because I just cherry-picked my changes on top of your PR, so when I run the tests, I didn't run the test you added here (sorry, my bad). This test is not valid anymore, since we cannot validate the schema since we have it only when we get the flow files. May you please remove `testValidateWithEL` then please? Thanks.

**Comment by mattyb149** [comment]
We don't need to get rid of those tests; rather I think if no EL is present, then we should try parsing the schema for validity. It will be reparsed in onTrigger() anyway (whether it has EL or not).

**Comment by mgaido91** [comment]
thanks @mattyb149. The test failures are not related to this PR. They are caused by 3ca7c3e7a19c7f496a214d9b93ace43b1507c9ec. I am investigating them. I am not sure if it is possible to re-trigger a build for this PR, thanks.

**Comment by mgaido91** [comment]
+1, locally it builds and passes all the tests

**Comment by pvillard31** [comment]
+1 as well, merging to master, thanks guys

**Pull request closed without merging**